assets/js/googlesitekit/modules/datastore/sharing-settings.test.js

```diff
@@ -0,0 +1,367 @@
+
+			it( 'receives management and sets it to the sharing settings modules', () => {
+				registry
+					.dispatch( CORE_MODULES )
+					.receiveGetSharingSettings( settingsWithoutManagement );
+
+				const state = {
+					...store.getState().sharingSettings,
+					...store.getState().savedSharingSettings,
+				};
+
+				const moduleSlugs = [ 'analytics', 'search-console' ];
+				const managementRoles = [ 'all_admins', 'owner' ];
+
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement(
+						moduleSlugs[ 0 ],
+						managementRoles[ 0 ]
+					);
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement(
+						moduleSlugs[ 1 ],
+						managementRoles[ 1 ]
+					);
+
+				const sharingSettingsWithManagement = moduleSlugs.reduce(
+					( sharingSettingsObj, moduleSlug, index ) => ( {
+						...sharingSettingsObj,
+						[ moduleSlug ]: {
+							...settingsWithoutManagement[ moduleSlug ],
+							management: managementRoles[ index ],
+						},
+					} ),
+					{}
+				);
+
+				expect( store.getState().sharingSettings ).toMatchObject( {
+					...state,
+					...sharingSettingsWithManagement,
+				} );
+			} );
+		} );
+
+		describe( 'setSharedRoles', () => {
+			const settingsWithoutRoles = {
+				'search-console': {
+					management: 'all_admins',
+				},
+				analytics: {
+					management: 'owner',
+				},
+				'pagespeed-insights': {
+					management: 'all_admins',
+				},
+			};
+
+			it( 'requires the moduleSlug param', () => {
+				expect( () => {
+					registry.dispatch( CORE_MODULES ).setSharedRoles();
+				} ).toThrow( 'moduleSlug is required' );
+			} );
+
+			it( 'requires the roles param', () => {
+				expect( () => {
+					registry
+						.dispatch( CORE_MODULES )
+						.setSharedRoles( 'analytics' );
+				} ).toThrow( 'roles must be an array of strings.' );
+			} );
+
+			it( 'receives roles and sets it to the sharing settings modules', () => {
+				registry
+					.dispatch( CORE_MODULES )
+					.receiveGetSharingSettings( settingsWithoutRoles );
+
+				const state = {
+					...store.getState().sharingSettings,
+					...store.getState().savedSharingSettings,
+				};
+
+				const moduleSlugs = [ 'analytics', 'search-console' ];
+				const sharedRoles = [ 'editor', 'subscriber' ];
+
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharedRoles( moduleSlugs[ 0 ], sharedRoles );
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharedRoles( moduleSlugs[ 1 ], sharedRoles );
+
+				const sharingSettingsWithSharedRoles = moduleSlugs.reduce(
+					( sharingSettingsObj, moduleSlug ) => ( {
+						...sharingSettingsObj,
+						[ moduleSlug ]: {
+							...settingsWithoutRoles[ moduleSlug ],
+							sharedRoles,
+						},
+					} ),
+					{}
+				);
+
+				expect( store.getState().sharingSettings ).toMatchObject( {
+					...state,
+					...sharingSettingsWithSharedRoles,
+				} );
+			} );
+		} );
+
+		describe( 'saveSharingSettings', () => {
+			it( 'does not require any params', () => {
```

Comment:
This test doesn't seem to be needed. If the action doesn't use params, then no need to test for it.

Reply:
We do have several tests that check for `does not require any params`. Shall we keep it, WDYT?

Reply:
Not sure about other tests, but here we don't need it.

Reply:
Sure, I will remove it.